### PR TITLE
Coyoneda docs - lets add some math ;)

### PR DESCRIPTION
**Pull request by coot**
I was reading through Coyoneda, and made some math notes. They might be useful for others, especially for ones that are learning PureScript and are not super familiar with category theory to have enough intuition to derive all the laws that one gets for free from the structures.

### CODE REVIEW

File: src/Data/Coyoneda.purs [hunk 1]
```diff
@@ -78,6 +102,21 @@ unCoyoneda :: forall f g a. (forall b. (b -> a) -> f b -> g a) -> Coyoneda f a -
 unCoyoneda f (Coyoneda e) = runExists (\(CoyonedaF k fi) -> f k fi) e
 
 -- | Lift a value described by the type constructor `f` to `Coyoneda f`.
+-- |
+-- | Note that for any `f` `liftCoyoneda` has a right inverse
```
Comment (paf31) [inline src/Data/Coyoneda.purs]: For any _functor_ `f`.

**Comment by paf31** [comment]
Looks good to me, just one minor nitpick. Thanks!

File: src/Data/Coyoneda.purs (hunk 1 above)
Comment (coot) [inline src/Data/Coyoneda.purs]: Good catch. For any f it has a left inverse.

ps. I will have more pr like this one.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**